### PR TITLE
YQL-18356: Refactor WideTopSort to be able use LLVM compilation

### PR DESCRIPTION
**Pull request by Darych**
### Changelog entry 

...

### Changelog category 

* New feature
* Experimental feature
* Improvement
* Performance improvement
* Bugfix 
* Backward incompatible change
* Documentation (changelog entry is not required)
* Not for changelog (changelog entry is not required)

### Additional information

...

### CODE REVIEW

File: ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp [hunk 1]
```diff
@@ -393,146 +369,115 @@ using TBase = TComputationValue<TSpillingSupportState<Sort, HasCount>>;
...
-                    }
-                    continue;
-                case EFetchResult::Yield:
-                    return EFetchResult::Yield;
-                case EFetchResult::Finish:
-                {
-                    if (!SpilledStates.empty()) {
-                        SwitchMode(EOperatingMode::Spilling, ctx);
-                        return EFetchResult::Yield;
+    bool IsReadyToContinue() {
```
Comment (lll-phill-lll) [inline ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp:403]: Nice, will do switch like this in GraceJoin

File: ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp [hunk 2]
```diff
@@ -393,146 +369,115 @@ using TBase = TComputationValue<TSpillingSupportState<Sort, HasCount>>;
...
+            case EOperatingMode::InMemory:
+               return true;
+            case EOperatingMode::Spilling:
+            {
+                if (!SpillState()) {
+                    return false;
+                }
+                ResetFields();
+                auto nextMode = (IsReadFromChannelFinished() ? EOperatingMode::ProcessSpilled : EOperatingMode::InMemory);
+                SwitchMode(nextMode);
```
Comment (lll-phill-lll) [inline ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp:414]: As far as I remember log was intended here after switching the modes, is it missing?

**Review by lll-phill-lll — DISMISSED** [review]

File: ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp (hunk 2 above)
Comment (lll-phill-lll) [reply ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp:414]: Oh, I see
Log conflicted here :(

**Review by lll-phill-lll — DISMISSED** [review]

File: ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp [hunk 3]
```diff
@@ -637,8 +589,47 @@ using TBase = TComputationValue<TSpillingSupportState<Sort, HasCount>>;
...
+        return static_cast<NUdf::TUnboxedValue*>(ptr);
+    }
+
+    void SealInMemory() {
+        if constexpr (!HasCount) {
+            static_assert (Sort);
+            // Remove placeholder for new data
+            Storage.resize(Storage.size() - Indexes.size());
+
+            Full.reserve(Storage.size() / Indexes.size());
```
Comment (lll-phill-lll) [inline ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp:607]: Do we have guarantees that `Indexes.size()` is not 0?

**Review by lll-phill-lll — APPROVED** [review]

File: ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp [hunk 4]
```diff
@@ -637,8 +589,47 @@ using TBase = TComputationValue<TSpillingSupportState<Sort, HasCount>>;
...
+            for (auto it = Storage.begin(); it != Storage.end(); it += Indexes.size()) {
+                Full.emplace_back(&*it);
+            }
+
+            std::sort(Full.rbegin(), Full.rend(), LessFunc);
+            return;
+        }
+
+        Free.clear();
+        Free.shrink_to_fit();
```
Comment (lll-phill-lll) [inline ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp:617]: Cppref says that `shrink_to_fit` is not obliged to actually shrink:
https://en.cppreference.com/w/cpp/container/vector/shrink_to_fit

As far as I know the only way to completely clear container is to swap it with empty container

File: ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp (hunk 3 above)
Comment (Darych) [reply ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp:607]: Yes. Indexes size is the number of input columns to the callable.

File: ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp (hunk 4 above)
Comment (Darych) [reply ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp:617]: Hm, I kept the previous implementation.
I think we should be fine for now: https://stackoverflow.com/questions/23502291/is-shrink-to-fit-the-proper-way-of-reducing-the-capacity-a-stdvector-to-its/23503995#23503995

**Pull request merged**